src/components/VideoPlayer.tsx

```diff
@@ -2,29 +2,50 @@ import React, { useEffect, useRef } from 'react';
 
 type IVideoPlayeProps = {
   srcVideo: string;
-  onVideoEnd: () => void;
+  onVideoEnd?: () => void;
 };
 
 const VideoPlayer = ({ srcVideo, onVideoEnd }: IVideoPlayeProps) => {
-  const videoRef: HTMLVideoElement = useRef();
+  const videoRef = useRef<HTMLVideoElement>(null);
 
   useEffect(() => {
-    if (onVideoEnd) {
-      const timerId = setInterval(() => {
-        if (videoRef.current.ended) {
-          onVideoEnd();
-          clearInterval(timerId);
-        }
-      }, 100);
+    if (!onVideoEnd) {
+      return;
     }
+    const timerId = setInterval(() => {
+      if (videoRef.current?.ended) {
+        onVideoEnd();
+        clearInterval(timerId);
+      }
+    }, 100);
+    return () => clearInterval(timerId);
+  }, [onVideoEnd]);
+
+  useEffect(() => {
+    videoRef.current.setAttribute('src', srcVideo);
+  }, [srcVideo]);
```

Comment:
Are we sure we need this useEffect, can't we just use `<source src={srcVideo} type="video/mp4" />` instead?